app/Controllers/Http/ProfilesController.ts

```diff
@@ -0,0 +1,101 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Profile from 'App/Models/Profile'
+import CreateProfile from 'App/Validators/CreateProfileValidator'
+import UpdateProfile from 'App/Validators/UpdateProfileValidator'
+
+export default class ProfilesController {
+    public async getUserProfile({ auth }: HttpContextContract) {
+        try {
+            const user = auth.use('api').user!
+            await user.load('profile')
+
+            if (user.profile) {
+                return {
+                    first_name: user.profile.firstName,
+                    last_name: user.profile.lastName,
+                    email: user.email,
+                    gender: user.profile.gender,
+                    dob: user.profile.dob
+                }
+            }
```

Comment:
Instead of `if` condition. It's better to check whether the profile exist or not. If not, then throw error.

Reply:
```suggestion
if (!user.profile) {
throw new UnprocessableException('Profile not found');
}
```

Reply:
Done